src/provide-context.test.js

```diff
@@ -0,0 +1,187 @@
+import * as React from "react";
+import provideContext from "./provide-context";
+import createResource from "./create-resource";
+import createSimpleResource from "./create-simple-resource";
+import { assertResourceShape } from "./__mocks__/assert-resource-shape";
+import { render, waitForElement } from "react-testing-library";
+
+describe("provideContext", () => {
+  it("is a curried function with an arity of two", () => {
+    expect(typeof provideContext).toBe("function");
+    expect(provideContext.length).toBe(2);
+
+    const result = provideContext(x => x);
+    expect(typeof result).toBe("function");
+    expect(result.length).toBe(1);
+  });
+
+  it("creates a resource", () => {
+    assertResourceShape(
+      provideContext(
+        x => x,
+        createSimpleResource(() => Promise.resolve("resolved"))
+      )
+    );
+  });
+
+  it("provides the return value of the provider to the selectState function", () => {
+    const value = "abc123";
+    const selectState = jest
+      .fn()
+      .mockImplementation((state, [authToken]) => state);
+    const resource = provideContext(
+      () => value,
```

Comment:
This is another piece that is incredible, it doesn't necessarily have to be a hook, it's "Just a Function™". (Although the best use case is likely a hook)